ui/pages/extensions.js

```diff
@@ -50,12 +51,59 @@ const styles = (theme) => ({
 
 const INITIAL_GRID_SIZE = { lg : 6, md : 12, xs : 12 };
 
-const Extensions = ({ classes, catalogVisibility, toggleCatalogContent }) => {
-
+const Extensions = ({ classes, catalogVisibility, toggleCatalogContent,  enqueueSnackbar, closeSnackbar }) => {
   const handleToggle = (e) => {
     e.stopPropagation();
     console.log(`Meshery Catalog: ${catalogVisibility ? "enabled" : "disabled"}`)
-    toggleCatalogContent({ catalogVisibility : !catalogVisibility });
+    toggleCatalogContent({ catalogVisibility : !catalogVisibility })
+  }
+
+  useEffect(() => {
+    dataFetch(
+      "/api/user/prefs",
+      {
+        method : "GET",
+        credentials : "include",
+      },
+      (result) => {
+        if (result) {
+          toggleCatalogContent({ catalogVisibility : !catalogVisibility });
+        }
+      },
+      err => console.error(err)
+    )
+  }, []);
+
+  useEffect(() => {
+    handleCatalogPreference();
+  }, [catalogVisibility])
+
+
+  const handleCatalogPreference = () => {
+    dataFetch(
+      "/api/user/prefs",
+      {
+        method : "POST",
+        credentials : "include",
+        body : JSON.stringify({ usersExtensionPreferences : { catalogVisibility } })
+      },
+      () => {
+        enqueueSnackbar(`Catalog Content was ${catalogVisibility ? "enab" : "disab"}led`,
+          { variant : 'success',
+            autoHideDuration : 4000,
+            action : (key) => (
+              <IconButton
+                key="close"
+                aria-label="Close"
+                color="inherit"
+                onClick={() => closeSnackbar(key)}
+              >
+                <CloseIcon />
+              </IconButton>
+            ),
+          });
+      },
+    )
```

Comment:
```suggestion
     },
     err => console.error(err). // or any error handler function of your choice
    )
```

Reply:
The reason for that error is the required number of parameters required for the `data-fetch` function.
It requires 4 parameters:
`url`, `options = {}`, `successFn`, `errorFn`

Currently, `errorFn` was not being passed, resulting in an undefined error while running the `data-fetch` function.

Reply:
@Chadha93 will you please check after adding the err handler function?

Reply:
Sure, @theBeginner86.